editor/shader_create_dialog.cpp

```diff
@@ -142,31 +142,66 @@ void ShaderCreateDialog::_create_new() {
 		code += vformat("shader_type %s;\n", mode_menu->get_text().replace(" ", "").camelcase_to_underscore());
 
 		if (current_template == 0) { // Default template.
-			code += "\n";
 			switch (current_mode) {
 				case Shader::MODE_SPATIAL:
-					code += "void fragment() {\n";
-					code += "\t// Place fragment code here.\n";
-					code += "}\n";
+					// `vertex()` and `light()` are commented out by default
+					// as these are used less often.
+					// Functions are ordered in the vertex-fragment-light order
+					// to follow established conventions.
+					code += R"(
+//void vertex() {
+	// Called for every vertex the material is visible on.
+//}
+
+void fragment() {
+	// Called for every pixel the material is visible on.
+}
+
+//void light() {
+	// Called for every light affecting the material.
+	// If you uncomment this function, you need to reimplement lighting equations
+	// to get an appearance similar to the default light shader.
+//}
+)";
 					break;
 				case Shader::MODE_CANVAS_ITEM:
-					code += "void fragment() {\n";
-					code += "\t// Place fragment code here.\n";
-					code += "}\n";
+					// `vertex()` and `light()` are commented out by default
+					// as these are used less often.
+					// Functions are ordered in the vertex-fragment-light order
+					// to follow established conventions.
+					code += R"(
+//void vertex() {
+	// Called for every vertex the material is visible on.
+//}
+
+void fragment() {
+	// Called for every pixel the material is visible on.
+}
+
+//void light() {
+	// Called for every light affecting the CanvasItem.
+	// If you uncomment this function, you need to reimplement lighting equations
+	// to get an appearance similar to the default light shader.
+//}
+)";
 					break;
 				case Shader::MODE_PARTICLES:
-					code += "void start() {\n";
-					code += "\t// Place start code here.\n";
-					code += "}\n";
-					code += "\n";
-					code += "void process() {\n";
-					code += "\t// Place process code here.\n";
-					code += "}\n";
+					code += R"(
+void start() {
+	// Called when a particle is spawned.
+}
+
+void process() {
+	// Called every frame on existing particles.
+}
+)";
 					break;
 				case Shader::MODE_SKY:
-					code += "void sky() {\n";
-					code += "\t// Place sky code here.\n";
-					code += "}\n";
+					code += R"(
+void sky() {
+	// Called for every texel in the sky texture.
```

Comment:
Is this description correct?

Reply:
Not quite, it is kind of complex because sky shaders are used for cubemaps as well. See the doc for a detailed explanation
https://docs.godotengine.org/en/latest/tutorials/shaders/shader_reference/sky_shader.html

I'll try to think of a comment we can add that is short, but accurate

Reply:
```suggestion
	// Called for every visible pixel in the sky background as well as all pixels in the radiance cubemap.
```